game-core/src/main/java/games/strategy/triplea/settings/SettingsWindow.java

```diff
@@ -43,10 +45,12 @@
 public enum SettingsWindow {
   INSTANCE;
 
+  private static final ImmutableList<SettingType> SETTING_TYPES = ImmutableList.copyOf(SettingType.values());
```

Comment:
Apart from the one syntactic sugar usage of `Iterable#forEach`, is this static field really needed?

Reply:
The real reason for this list is to act as a `tab index` -> `SettingType` map.  That functionality is used in `getSelectedSelectionComponents()`:

```java
return Optional.of(getSettingsByType(SETTING_TYPES.get(selectedTabIndex)).stream()
    .map(selectionComponentsBySetting::get)
    .collect(Collectors.toList()));
```

Not sure if converting this to a `Map<Integer, SettingType>` named `SETTING_TYPES_BY_TAB_INDEX` would make that clearer?

Reply:
But `SettingType.values()` should always return the same array with an consistent order, so to map an index to a setting the array access operator (`[]`) could be used instead, or am I missing something obvious?

Reply:
No, you're correct.  However, while `SettingType#values()` returns an array with the same _elements_ on each call, the array itself is newly-created each time (presumably to act as a defensive copy).  Since `getSelectedSelectionComponents()` may be invoked numerous times while this window is displayed, that seems like it could lead to a lot of excessive, and unnecessary, allocations for something that is effectively constant for the lifetime of the JVM.

c2693e0c1